hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/EightToSevenDowngradeHandler.java

```diff
@@ -79,17 +96,60 @@ public Map<ConfigProperty, String> downgrade(HoodieWriteConfig config, HoodieEng
     }
 
     downgradePartitionFields(config, context, upgradeDowngradeHelper, tablePropsToAdd);
+    downgradeMetadataPartitions(metaClient, context, tablePropsToAdd);
     return tablePropsToAdd;
   }
 
-  private static void downgradePartitionFields(HoodieWriteConfig config, HoodieEngineContext context, SupportsUpgradeDowngrade upgradeDowngradeHelper,
+  private static void downgradePartitionFields(HoodieWriteConfig config,
+                                               HoodieEngineContext context,
+                                               SupportsUpgradeDowngrade upgradeDowngradeHelper,
                                                Map<ConfigProperty, String> tablePropsToAdd) {
     HoodieTableConfig tableConfig = upgradeDowngradeHelper.getTable(config, context).getMetaClient().getTableConfig();
     String keyGenerator = tableConfig.getKeyGeneratorClassName();
     String partitionPathField = config.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key());
-    if (keyGenerator != null && partitionPathField != null
-        && (keyGenerator.equals(KeyGeneratorType.CUSTOM.getClassName()) || keyGenerator.equals(KeyGeneratorType.CUSTOM_AVRO.getClassName()))) {
+    if (keyGenerator != null
+        && partitionPathField != null
+        && (keyGenerator.equals(KeyGeneratorType.CUSTOM.getClassName())
+          || keyGenerator.equals(KeyGeneratorType.CUSTOM_AVRO.getClassName()))) {
       tablePropsToAdd.put(HoodieTableConfig.PARTITION_FIELDS, tableConfig.getPartitionFieldProp());
     }
   }
+
+  static void downgradeMetadataPartitions(HoodieTableMetaClient metaClient,
+                                          HoodieEngineContext context,
+                                          Map<ConfigProperty, String> tablePropsToAdd) {
+    metaClient.reloadActiveTimeline();
+    StoragePath basePath = metaClient.getBasePath();
+
+    HoodieTableMetaClient mdtMetaClient = HoodieTableMetaClient.builder()
+        .setConf(HadoopFSUtils.getStorageConfWithCopy(
+            (Configuration) context.getStorageConf().unwrap()))
+        .setBasePath(
+            new Path(basePath.toString(),
+                HoodieTableMetaClient.METADATA_TABLE_FOLDER_PATH).toString())
+        .setLoadActiveTimelineOnLoad(true)
+        .build();
+
+    // Delete partitions.
+    deleteMetadataPartition(context, metaClient.getStorage(), mdtMetaClient);
+
+    // No matter if partitions have been removed, block their creation.
+    tablePropsToAdd.put(TABLE_METADATA_PARTITIONS, "");
```

Comment:
this config could have other values (metadata partitions) that are supported in previous versions e.g. files,record_index,column_stats etc. We should not be setting it to empty. Shouldn't we only remove the newer partitions that are supported since 1.0?

Reply:
Yeah, I will fix.